rotkehlchen/chain/evm/decoding/efp/decoder.py

```diff
@@ -0,0 +1,158 @@
+
+    def _get_address_for_slot(self, slot: int) -> 'ChecksumEvmAddress | None':
+        """Get address associated with the specified slot.
+        Returns an address or None if unable to match an address to this slot."""
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if (address := globaldb_get_unique_cache_value(
+                    cursor=cursor,
+                    key_parts=(CacheType.EFP_SLOT_ADDRESS, str(slot)),
+            )) is not None:
+                return string_to_evm_address(address)
+
+        try:
+            if (address := to_checksum_address(self.evm_inquirer.contracts.contract(self.list_records_contract).call(  # noqa: E501
+                node_inquirer=self.evm_inquirer,
+                method_name='getListUser',
+                arguments=[slot],
+            ))) == ZERO_ADDRESS:
+                log.error(f'Failed to retrieve address for EFP slot {slot} on {self.evm_inquirer.chain_id.label()}')  # noqa: E501
+                return None
+        except RemoteError as e:
+            log.error(f'Failed to retrieve address for EFP slot {slot} on {self.evm_inquirer.chain_id.label()} due to {e!s}')  # noqa: E501
+            return None
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.EFP_SLOT_ADDRESS, str(slot)),
+                value=address,
+            )
+        return address
+
+    def _decode_list_op_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode EFP list operation events."""
+        if context.tx_log.topics[0] != LIST_OP_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        if (address := self._get_address_for_slot(
+                slot=(slot := int.from_bytes(context.tx_log.topics[1])),
+        )) is None:
+            log.error(
+                f'Unable to decode EFP list op event due to unknown slot {slot} '
+                f'in transaction {context.transaction}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        if not self.base.is_tracked(address):
+            log.debug(
+                f'Skipping EFP list op event for untracked address {address} '
+                f'in transaction {context.transaction}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        # Decode list operation data. See https://docs.ethfollow.xyz/design/list-ops/
+        data = context.tx_log.data[64:]
+        op_version = int.from_bytes(data[0:1])
+        op_code = int.from_bytes(data[1:2])
+        record_version = int.from_bytes(data[2:3])
+        record_type = int.from_bytes(data[3:4])
+        if op_version != 1 or op_code not in [1, 2, 3, 4] or record_version != 1 or record_type != 1:  # noqa: E501
+            log.error(
+                f'Found unsupported EFP list operation in transaction {context.transaction}. '
```

Comment:
chain id is probably in this log too.

Reply:
`context.transaction` includes the chain id via `EvmTransaction`'s `__str__` function: https://github.com/rotki/rotki/blob/develop/rotkehlchen/types.py#L419